### PR TITLE
Username validation at api level#827

### LINKED ISSUE #827: [Feature Request|Bug] Enforcing Username Uniqueness and Character Restrictions for Account Management
Usernames should be unique; currently, admin and Admin are two separate accounts.
Usernames should not contain spaces or percentage signs - a user with spaces cannot be deleted, and further special characters need to be checked.

### PR DESCRIPTION
**Pull request by mannam11**
Added username validation at API level.

Closes #827 

## Checklist:

- [x] I have read the [Contribution Guidelines](https://github.com/Stirling-Tools/Stirling-PDF/blob/main/CONTRIBUTING.md)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings

## Contributor License Agreement

By submitting this pull request, I acknowledge and agree that my contributions will be included in Stirling-PDF and that they can be relicensed in the future under the MPL 2.0 (Mozilla Public License Version 2.0) license.

(This does not change the general open-source nature of Stirling-PDF, simply moving from one license to another license)

### CODE REVIEW

**Comment by mannam11** [comment]
@Frooodle I have added the API level validation. Can you please review this PR.

**Comment by Frooodle** [comment]
Logic should be added to other functions which can pdate username such as
changeUsername
changeUsernameAndPassword
register

**Comment by mannam11** [comment]
> Logic should be added to other functions which can pdate username such as changeUsername changeUsernameAndPassword register

working on it.

**Comment by Frooodle** [comment]
Register only had API side not UI it's internal thing

File: src/main/resources/templates/addUsers.html [hunk 1]
```diff
@@ -37,37 +37,43 @@ <h2 class="text-center" th:text="#{adminUserSettings.header}">Admin User Control
...
+								<span th:text="#{usernameExistsMessage}">Default message if not found</span>
+							</div>
+							<div th:case="'invalidUsername'" class="alert alert-danger">
+								<span th:text="#{invalidUsernameMessage}">Default message if not found</span>
+							</div>
+						</div>
+						<form action="/api/v1/user/admin/saveUser" method="post">
+					        <div class="mb-3">
+					            <label for="username" th:text="#{username}">Username</label>
+					            <input type="text" class="form-control" name="username" pattern="[a-zA-Z0-9]+" title="Username must only contain letters and numbers and no spaces or special characters" required>
```
Comment (sbplat) [inline src/main/resources/templates/addUsers.html:53]: This should use the translation attribute.

**Pull request closed without merging**